tests/common-tests.h

```diff
@@ -6364,6 +6384,375 @@ TEST_CASE_METHOD(common_tests, "Into XML vector with several fetches", "[core][x
+            }
+
+            // Update original blob and insert new db-entry (must not change previous entry)
+            const int second_id = first_id + 1;
+            write_blob.trim(0);
+            static_assert(sizeof(dummy_data) >= 15 + 5, "Underlying assumption violated");
+            write_blob.write_from_start(dummy_data + 15, 5);
+
+            sql << "insert into soci_test (id, b) values (:id, :b)", soci::use(second_id), soci::use(write_blob);
+
+            // First, check that the original entry has not been changed
+            sql << "select b from soci_test where id = :id", soci::use(first_id), soci::into(read_blob);
+            CHECK(read_blob.get_len() == 10);
+
+            // Then check new entry can be read
+            sql << "select b from soci_test where id = :id", soci::use(second_id), soci::into(read_blob);
+
+            bytes_read = read_blob.read_from_start(buf, sizeof(buf));
+            CHECK(bytes_read == read_blob.get_len());
+            CHECK(bytes_read == 5);
+            for (std::size_t i = 0; i < bytes_read; ++i)
+            {
+                CHECK(buf[i] == dummy_data[i + 15]);
+            }
+        }
+        SECTION("Binary data")
+        {
+            const std::uint8_t binary_data[12] = {0, 1, 2, 3, 4, 5, 6, 7, 22, 255, 250 };
+
+            soci::blob write_blob(sql);
+
+            std::size_t bytes_written = write_blob.write_from_start(reinterpret_cast<const char *>(binary_data), sizeof(binary_data));
+            CHECK(bytes_written == sizeof(binary_data));
+
+            sql << "insert into soci_test (id, b) values (1, :b)", soci::use(write_blob);
+
+            soci::blob read_blob(sql);
+
+            sql << "select b from soci_test where id = 1", soci::into(read_blob);
+
+            CHECK(read_blob.get_len() == sizeof(binary_data));
+
+            std::uint8_t buf[20];
+            std::size_t bytes_read = read_blob.read_from_start(reinterpret_cast<char *>(buf), sizeof(buf));
+
+            CHECK(bytes_read == sizeof(binary_data));
+            for (std::size_t i = 0; i < sizeof(binary_data); ++i)
+            {
+                CHECK(buf[i] == binary_data[i]);
+            }
+        }
+        SECTION("Rowset Blob recognition")
+        {
+            soci::blob blob(sql);
+
+            // Write and retrieve blob from/into database
+            int id = 1;
+            sql << "insert into soci_test (id, b) values(:id, :b)", soci::use(id), soci::use(blob);
+
+            soci::rowset< soci::row > rowSet = sql.prepare << "select id, b from soci_test";
+            bool containedData = false;
+            for (auto it = rowSet.begin(); it != rowSet.end(); ++it)
+            {
+                containedData = true;
+                const soci::row &currentRow = *it;
+
+                soci::data_type type = currentRow.get_properties(0).get_data_type();
+                soci::data_type expectedType = tc_.get_backend() != Backend::Oracle ? soci::dt_integer : soci::dt_long_long;
```

Comment:
> It's really not clear why do we have this difference

Neither do I - this was not introduced by me :shrug: 

> but it would be even better to get rid of it instead.

I agree, but I'm wondering whether that may break something else :eyes: 

> Maybe the simplest would be to just not check the type of the ID column, after all this is not what this test is about.

Probably, yeah :+1: